packages/dev/gui/src/2D/controls/rectangle.ts

```diff
@@ -181,17 +194,17 @@ export class Rectangle extends Container {
         const height = this._currentMeasure.height - offset * 2;
 
         let radius = {
-            x: Math.min(height / 2, Math.min(width / 2, this._cornerRadiusFree.x)),
-            y: Math.min(height / 2, Math.min(width / 2, this._cornerRadiusFree.y)),
-            z: Math.min(height / 2, Math.min(width / 2, this._cornerRadiusFree.z)),
-            w: Math.min(height / 2, Math.min(width / 2, this._cornerRadiusFree.w))
-        }
+            x: Math.min(height / 2, Math.min(width / 2, this._cornerRadius[0])),
+            y: Math.min(height / 2, Math.min(width / 2, this._cornerRadius[1])),
+            z: Math.min(height / 2, Math.min(width / 2, this._cornerRadius[2])),
+            w: Math.min(height / 2, Math.min(width / 2, this._cornerRadius[3])),
+        };
         radius = {
             x: Math.abs(radius.x),
             y: Math.abs(radius.y),
             z: Math.abs(radius.z),
-            w: Math.abs(radius.w)
-        }
+            w: Math.abs(radius.w),
+        };

```

Comment:
is it possible to do that in one computation? The same reason as above - GC is triggered and can be avoided

Reply:
The object can be cached on the class level (private _cachedRadius...) and then completely avoid GC.